lambdas/service/app.py

```diff
@@ -257,14 +256,16 @@
             'name': 'Repository',
             'description': 'Access to data files in the underlying repository'
         },
-        {
-            'name': 'DSS',
-            'description': 'Access to files maintained in the Data Store'
-        },
-        {
-            'name': 'DRS',
-            'description': 'DRS-compliant proxy of the underlying repository'
-        },
+        *([
+              {
+                  'name': 'DSS',
+                  'description': 'Access to files maintained in the Data Store'
+              },
+              {
+                  'name': 'DRS',
+                  'description': 'DRS-compliant proxy of the underlying repository'
+              }] if config.is_dss_enabled() else [
```

Comment:
This is a bit brittle. Can we delete all unused tags in a post-processing step?

Reply:
What if I add a test that asserts there are no unused tags?

Reply:
It's possible that the decision which endpoints to enable—and therefore which tags are in use—is not made until runtime so a test can only approximate this. Given that I don't see any significant reduction of complexity if this were done as a test, I'd prefer the run time solution I proposed.

Reply:
I'm still sort of confused about why a post-processing step is desirable. If we have unused tags, then something in the code is wrong. Shouldn't we see the unused tags so that we can address the underlying issue? Otherwise we may end up with dead tags that aren't removed from the code because we never notice it on the Swagger page. A post-processing step feel like defensive programming to me.

Reply:
> If we have unused tags, then something in the code is wrong.

I don't think so. If an endpoint is disabled via some condition, and that endpoint is the only endpoint using a particular tag, that tag ends up not being used. Nothing wrong with that. Maybe you are not aware of the `enabled` keyword arg to the `app.route` decorator.